x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -285,37 +338,174 @@ void persistStateToClusterState(DataFrameTransformState state,
         private final String transformId;
+        private final Auditor<DataFrameAuditMessage> auditor;
+        private final DataFrameTransformTask transformTask;
+        private final Map<String, String> fieldMappings;
+        private final DataFrameTransformConfig transformConfig;
+        private volatile DataFrameTransformProgress progress;
         private volatile DataFrameIndexerTransformStats previouslyPersistedStats = null;
+        private final AtomicInteger failureCount;
         // Keeps track of the last exception that was written to our audit, keeps us from spamming the audit index
         private volatile String lastAuditedExceptionMessage = null;
-        private Map<String, String> fieldMappings = null;
 
-        private DataFrameTransformConfig transformConfig = null;
-
-        public ClientDataFrameIndexer(String transformId, DataFrameTransformsConfigManager transformsConfigManager,
-                                      DataFrameTransformsCheckpointService transformsCheckpointService,
-                                      AtomicReference<IndexerState> initialState, Map<String, Object> initialPosition, Client client,
-                                      Auditor<DataFrameAuditMessage> auditor) {
-            super(threadPool.executor(ThreadPool.Names.GENERIC), auditor, initialState, initialPosition,
-                new DataFrameIndexerTransformStats(transformId));
+        ClientDataFrameIndexer(String transformId,
+                               DataFrameTransformsConfigManager transformsConfigManager,
+                               DataFrameTransformsCheckpointService transformsCheckpointService,
+                               AtomicReference<IndexerState> initialState,
+                               Map<String, Object> initialPosition,
+                               Client client,
+                               Auditor<DataFrameAuditMessage> auditor,
+                               DataFrameIndexerTransformStats initialStats,
+                               DataFrameTransformConfig transformConfig,
+                               Map<String, String> fieldMappings,
+                               DataFrameTransformProgress transformProgress,
+                               DataFrameTransformTask parentTask) {
+            super(parentTask.threadPool.executor(ThreadPool.Names.GENERIC), auditor, initialState, initialPosition, initialStats);
             this.transformId = transformId;
             this.transformsConfigManager = transformsConfigManager;
             this.transformsCheckpointService = transformsCheckpointService;
             this.client = client;
+            this.auditor = auditor;
+            this.transformConfig = transformConfig;
+            this.transformTask = parentTask;
+            this.fieldMappings = fieldMappings;
+            this.progress = transformProgress;
+            this.failureCount = new AtomicInteger(0);
```

Comment:
neither here nor in the builder essential parameters are not checked `!= null`

Reply:
Definitely, the constructor should throw on `null` params. Will fix soon.